File: Programmers/Lv3/최고의 집합.py
def solution(n, s):
    answer = []
    if s < n == 0:
        return [-1]

    num = s // n
    if s // n == 0:
        return [-1]

    rest = s % n

    for i in range(n):
        answer.append(num)
        # 수만큼 일단 half 값 넣기

    if rest != 0:
        for i in range(len(answer)):
            answer[i] += 1
            rest -= 1
            if rest == 0:
                break
    answer.sort()
    return answer

File: Programmers/Lv3/최고의 집합.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(int n, int s) {
    if(s < n){
        return {-1};
    }
    
    int num = s / n;
    int rest = s % n;
    vector<int> res(n, num);
    
    if(s / num == 0){
        return {-1};
    } else {
        int idx = 0;
        while(rest > 0){
            res[idx]++;
            idx++;
            rest--;
        }
        sort(res.begin(), res.end());
        return res;
    }
}